content/en/docs/components/trainer/user-guides/local-execution-mode/local_process.md

```diff
@@ -0,0 +1,535 @@
+
+```bash
+python -m venv --help
+```
+
+### Package Installation Errors
+
+**Problem**: Required packages fail to install in the virtual environment.
+
+**Solution**:
+- Check your internet connection
+- Verify that package names are correct
+- Use `packages_to_install` to explicitly specify packages
+- Ensure pip is up to date: `pip install --upgrade pip`
+
+### Jobs Not Cleaning Up
+
+**Problem**: Virtual environments remain after job completion.
+
+**Solution**: Verify that `cleanup_venv=True` in your config, or manually delete jobs:
+
+```python
+client.delete_job(job_name)
+```
+
+### Permission Errors
+
+**Problem**: Permission denied when creating virtual environments.
+
+**Solution**: Ensure you have write permissions to the temp directory. On Unix-like systems, check:
+
+```bash
+ls -ld /tmp
+```
+
+### Debug Mode
+
+Enable debug logging for detailed execution information:
+
+```python
+import logging
+logging.basicConfig(level=logging.DEBUG)
+
+backend_config = LocalProcessBackendConfig(cleanup_venv=False)
+client = TrainerClient(backend_config=backend_config)
+```
+
+## Limitations
+
+- **Single Machine Only**: Only runs on local machine, no distributed training across multiple nodes. The `num_nodes` parameter is ignored.
+- **CustomTrainer Only**: Does not support BuiltinTrainer configurations.
+- **No GPU Scheduling**: Cannot manage GPU allocation across multiple jobs.
+- **Process Isolation**: Jobs are isolated by virtual environment, not containers.
+- **Limited Scaling**: Not suitable for large-scale production training.
+- **System Dependencies**: Training code must be compatible with your local Python environment and operating system.
+
+## Comparison with Container Backend
+
+| Feature | Local Process Backend | Container Backend |
+|---------|----------------------|-------------------|
+| **Setup** | No additional software required | Requires Docker or Podman |
+| **Isolation** | Python virtual environments | Full container isolation |
+| **Multi-node Training** | Not supported | Supported |
+| **Startup Time** | Fast (seconds) | Slower (image pull + container start) |
+| **Resource Isolation** | Limited (shared system) | Strong (cgroups, namespaces) |
+| **GPU Support** | Direct system access | NVIDIA Container Toolkit required |
+| **Best For** | Quick prototyping, development | Production, distributed training |
+
+## Migration Guide
+
+### From Kubernetes Backend
+
+The same training code works in both backends:
+
+```python
+# Before (Kubernetes)
+from kubeflow.trainer.backends.kubernetes import KubernetesBackendConfig
+backend_config = KubernetesBackendConfig(namespace="default")
+
+# After (Local Process)
+from kubeflow.trainer.backends.localprocess import LocalProcessBackendConfig
+backend_config = LocalProcessBackendConfig()
+
+# Same trainer, same train() call!
+client = TrainerClient(backend_config=backend_config)
+trainer = CustomTrainer(func=train_model)
+job_name = client.train(trainer=trainer)
+```
```

Comment:
I think, we should have separate page which explains the Job management and switch between backends.
Maybe you can have overview section in `/local-execution-mode/overview.md` ?
Similar as here: https://www.kubeflow.org/docs/components/trainer/operator-guides/job-scheduling/overview/